samples/tab-approval/nodejs/Readme.md

```diff
@@ -0,0 +1,155 @@
+    - `ChatMessage.Send`
+    - `Chat.ReadWrite`
+    - `TeamsActivity.Send`
+    - `TeamsAppInstallation.ReadForUser.All`.
+
+**Note** Your need to add `TeamsActivity.Send` and `Directory.Read.All` as Application level permissions too.
+
+-   Click on Add permissions. Please make sure to grant the admin consent for the required permissions.
+13. Navigate to **Authentication**
+    If an app hasn't been granted IT admin consent, users will have to provide consent the first time they use an app.
+    Set a redirect URI:
+    * Select **Add a platform**.
+    * Select **web**.
+    * Enter the **redirect URI** for the app in the following format: `https://{Base_Url}/Auth/end`, `https://{Base_Url}/Auth/Start`. This will be the page where a successful implicit grant flow will redirect the user.
+    Again
+	* Select **Single page application**.
+	* Enter the **redirect URI** for the app in the following format: `https://{Base_Url}/tabAuth`
+	
+    Enable implicit grant by checking the following boxes:  
+    ✔ ID Token  
+    ✔ Access Token  
+14.  Navigate to the **Certificates & secrets**. In the Client secrets section, click on "+ New client secret". Add a description(Name of the secret) for the secret and select “Never” for Expires. Click "Add". Once the client secret is created, copy its value, it need to be placed in the appsettings.json.
+
+### 2. Run your bot sample
+1) Clone the repository
+
+    ```bash
+    git clone https://github.com/OfficeDev/Microsoft-Teams-Samples.git
+    ```
+
+2) In the folder where repository is cloned navigate to `samples/tab-approval/nodejs`
+
+3) Install node modules
+
+   Inside node js folder, open your local terminal and run the below command to install node modules. You can do the same in Visual studio code terminal by opening the project in Visual studio code 
+
+    ```bash
+    npm install
+    ```
+4) Run ngrok - point to port 3978
+
+    ```bash
+    ngrok http -host-header=rewrite 3978
+    ```
+5) Update the `.env` configuration file in your project folder for the bot to use the `ClientId` and `ClientSecret`, `TenantId` with your tenant id. For e.g., your ngrok url. (Note the ClientId is the AppId created in step 1 (Setup for Bot), the ClientSecret is referred to as the "client secret" in step 1 (Setup for Bot) and you can always create a new client secret anytime.)
+
+6) Run your app
+
+    ```bash
+    npm start
+    ```
+7) Manually update the manifest.json
+    - Edit the `manifest.json` contained in the  `appPackage/` folder to replace with your MicrosoftAppId (that was created in step1.1 and is the same value of MicrosoftAppId in `.env` file) *everywhere* you see the place holder string `{MicrosoftAppId}` (depending on the scenario the Microsoft App Id may occur multiple times in the `manifest.json`)
+    - Zip up the contents of the `appPackage/` folder to create a `manifest.zip`
+    - Upload the `manifest.zip` to Teams (in the left-bottom *Apps* view, click "Upload a custom app")
```

Comment:
Note: App should be installed for user's manager also to get task approval notification.